packages/action-listener-middleware/src/index.ts

```diff
@@ -334,6 +351,64 @@ export function createActionListenerMiddleware<
     return true
   }
 
+  const notifyListener = async (
+    entry: ListenerEntry<unknown, Dispatch<AnyAction>>,
+    action: AnyAction,
+    api: MiddlewareAPI,
+    getOriginalState: () => S,
+    currentPhase: MiddlewarePhase
+  ) => {
+    const internalTaskController = new AbortController()
+    const take = createTakePattern(addListener, internalTaskController.signal)
+    const condition: ConditionFunction<S> = (predicate, timeout) => {
+      return take(predicate, timeout).then(Boolean)
+    }
+    const delay = createDelay(internalTaskController.signal)
+    const fork = createFork(internalTaskController.signal)
+    const pause: (val: Promise<any>) => Promise<any> = createPause(
+      internalTaskController.signal
+    )
+    try {
+      entry.pendingSet.add(internalTaskController)
+      await Promise.resolve(
+        entry.listener(action, {
+          ...api,
+          getOriginalState,
+          condition,
+          take,
+          delay,
+          pause,
+          currentPhase,
+          extra,
+          signal: internalTaskController.signal,
+          fork,
+          unsubscribe: entry.unsubscribe,
+          subscribe: () => {
+            listenerMap.set(entry.id, entry)
+          },
+          cancelPrevious: () => {
+            entry.pendingSet.forEach((controller, _, set) => {
```

Comment:
:hand: I actually have had a concern with `cancelPrevious()` for a while now.

As implemented, it's really not "cancel _previous_", it's "cancel _all other instances regardless of which came first".  Which is a valid thing to do, but it's not semantically the same :)

It almost makes me wonder if we should have separate `cancelPrevious()` and `cancelOthers()` functions?  Or maybe rename this?